lib/ts/recipe/thirdParty/providers/facebook.ts

```diff
@@ -0,0 +1,121 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import { TypeProvider, TypeProviderGetResponse } from "../types";
+import axios from "axios";
+import { validateTheStructureOfUserInput } from "../../../utils";
+import Receipe from "../recipe";
+
+type TypeThirdPartyProviderFacebookConfig = {
+    clientId: string;
+    clientSecret: string;
+    scope?: string[];
+};
+
+const InputSchemaTypeThirdPartyProviderFacebookConfig = {
+    type: "object",
+    properties: {
+        clientId: {
+            type: "string",
+        },
+        clientSecret: {
+            type: "string",
+        },
+        scope: {
+            type: "array",
+            items: {
+                type: "string",
+            },
+        },
+        required: ["clientId", "clientSecret"],
+        additionalProperties: false,
+    },
+};
+
+export default function Facebook(config: TypeThirdPartyProviderFacebookConfig): TypeProvider {
+    validateTheStructureOfUserInput(
+        config,
+        InputSchemaTypeThirdPartyProviderFacebookConfig,
+        "thirdparty recipe, provider facebook",
+        Receipe.RECIPE_ID
+    );
+    const id = "facebook";
+
+    async function get(redirectURI: string, authCodeFromRequest: string | undefined): Promise<TypeProviderGetResponse> {
+        let accessTokenAPIURL = "https://graph.facebook.com/v9.0/oauth/access_token";
+        let accessTokenAPIParams: { [key: string]: string } = {
+            client_id: config.clientId,
+            client_secret: config.clientSecret,
+            redirect_uri: redirectURI,
+        };
+        if (authCodeFromRequest !== undefined) {
+            accessTokenAPIParams.code = authCodeFromRequest;
+        }
+        let authorisationRedirectURL = "https://www.facebook.com/v9.0/dialog/oauth";
+        let scopes = ["email"];
+        if (config.scope !== undefined) {
+            scopes.push(...config.scope);
+            scopes = Array.from(new Set(scopes));
+        }
+        let authorizationRedirectParams: { [key: string]: string } = {
+            scope: scopes.join(" "),
+            response_type: "code",
+            redirect_uri: redirectURI,
+            client_id: config.clientId,
+        };
+
+        async function getProfileInfo(accessTokenAPIResponse: {
+            access_token: string;
+            expires_in: number;
+            token_type: string;
+        }) {
+            let accessToken = accessTokenAPIResponse.access_token;
+            let response = await axios({
+                method: "get",
+                url: "https://graph.facebook.com/me",
+                params: {
+                    access_token: accessToken,
+                    fields: "id,email",
+                    format: "json",
+                },
+            });
+            let userInfo = response.data;
+            let id = userInfo.id;
+            let email = userInfo.email;
```

Comment:
Shouldn't you check if the email is undefined or not? And accordingly, return the `email: {}` object? (Same applies to all providers)